server/src/main/java/com/google/fhir/gateway/FhirUtil.java

```diff
@@ -109,7 +109,13 @@ public static String checkIdOrFail(String idPart) {
   }
 
   public static String extractLogicalId(Resource resource) {
-    return resource.getResourceType() + "/" + resource.getIdElement().getIdPart();
+    return resource.getIdElement().hasVersionIdPart()
```

Comment:
Given that `extractLogicalId` is used elsewhere too, let's make this `version` branch explicit; i.e., create an overloaded `extractLogicalId` with an extra `boolean` that indicates whether to include version or not. The non-overloaded `extractLogicalId` should call that function with `false`.